syntax = "proto2";
option optimize_for = CODE_SIZE;

import "google/protobuf/descriptor.proto";
import "blickfeld/config/generate.proto";
import "blickfeld/config/secure.proto";

package blickfeld.protocol;

extend google.protobuf.OneofOptions {
	optional bool optional_one_of = 50006 [default = false];
}

extend google.protobuf.FieldOptions {
	optional double d_min = 50000 [default = -inf];
	optional double d_max = 50001 [default = inf];
	optional sint32 length = 50004 [default = 2147483647];
	optional string regex = 50005 [default = ".*"];
	optional bool optional = 50006 [default = false];
	optional bool allow_sparse = 50007 [default = false];
	optional sint32 min_length = 50008 [default = 0];
	optional sint32 max_length = 50009 [default = 2147483647];
	optional uint64 legacy_field_id = 50010;
	optional string unit = 50011; // Unit of field value
	optional string ui_unit = 50012; // Unit of field value in user interfaces (UIs).
	optional double ui_scale = 50013 [default = 1]; // Scaling of field value in user interfaces (UIs). Example: rad-to-degree conversions.
	optional uint32 ui_decimal_places = 50014 [default = 0];
}

extend google.protobuf.MessageOptions {
	optional string e_desc = 60000 [default = "No additional error description available."]; // Error description
	optional string help = 60001 [default = ""]; // Help description
	optional blickfeld.protocol.config.Secure secure = 60002;
	optional blickfeld.protocol.config.Generate generate = 60003;
}
